hedera-node/src/main/java/com/hedera/services/store/models/Token.java

```diff
@@ -102,27 +124,237 @@
 	private int decimals;
 	private long autoRenewPeriod;
 	private long lastUsedSerialNumber;
+	static Predicate<Key> REMOVES_ADMIN_KEY = ImmutableKeyUtils::signalsKeyRemoval;
+	private JKey feeScheduleKey;
 	private List<CustomFee> customFees;
 
 	public Token(Id id) {
 		this.id = id;
 	}
 
+	/**
+	 * Updates the token properties if appropriate and if the transaction body complies with business rules for the update operation.
+	 * <p><b>Candidates of the update operation are the bellow token properties:</b></p>
+	 * <li>Keys (Freeze, KYC, Supply, Wipe, and Fee Schedule)</li>
+	 * <li>Symbol</li>
+	 * <li>Name</li>
+	 * <li>Memo</li>
+	 * <li>Expiry</li>
+	 * <li>Treasury account</li>
+	 *
+	 * @param changes             the transaction body containing the necessary data for token update operation
+	 * @param newAutoRenewAccount the new auto-renew account, which will replace the current auto-renew account if appropriate
+	 * @param newTreasury         the new Treasury account of the token
+	 * @param newTreasuryRel      the relationship between the token and the new Treasury account
+	 * @param validator           instance of {@link com.hedera.services.txns.validation.OptionValidator} (used to validate the auto-renew period if present in the transaction body)
+	 */
+	public void update(
```

Comment:
nit: do not pass the whole UpdateTransactionBody, but only the required properties. This way we will "terminate" the use of the GRPC model at the start of this method